qiita_db/sql_connection.py

```diff
@@ -1012,6 +1014,20 @@ def execute_fetchindex(self, idx=-1):
         """
         return self.execute()[idx]
 
+    def _cmds_executor(self, cmds, func_str):
+        error_msg = []
+        for cmd in cmds:
+            try:
+                cmd()
+            except Exception as e:
+                error_msg.append(str(e))
+        self._post_commit_funcs = []
+        self._post_rollback_funcs = []
+        if error_msg:
+            raise RuntimeError(
```

Comment:
I can't find a test that checks for a case where :hankey: hits the fan, can you add one?